code/game/machinery/digital_clock.dm

```diff
@@ -0,0 +1,156 @@
+/obj/machinery/digital_clock
+	name = "digital clock"
+	desc = "An ultra-futuristic, sleek, advanced, next-gen normal digital clock that tells the time. Powered by bluespace. Despite supposedly being better in every way to classic clocks, it just doesn't feel the same. They just don't make them the way they used to..."
+	icon_state = "digital_clock_base"
+	icon = 'icons/obj/digital_clock.dmi'
+	verb_say = "beeps"
+	verb_ask = "bloops"
+	verb_exclaim = "blares"
+	density = FALSE
+	layer = ABOVE_WINDOW_LAYER
+	custom_materials = list(/datum/material/iron = SHEET_MATERIAL_AMOUNT * 7, /datum/material/glass = SHEET_MATERIAL_AMOUNT * 4)
+
+/obj/item/wallframe/digital_clock
+	name = "digital clock frame"
+	desc = "Used to build digital clocks, just secure to the wall."
+	icon_state = "digital_clock_base"
+	icon = 'icons/obj/digital_clock.dmi'
+	custom_materials = list(/datum/material/iron = SHEET_MATERIAL_AMOUNT * 7, /datum/material/glass = SHEET_MATERIAL_AMOUNT * 4)
+	result_path = /obj/machinery/digital_clock
+	pixel_shift = 28
+
+/obj/machinery/digital_clock/wrench_act_secondary(mob/living/user, obj/item/tool)
+	. = ..()
+	balloon_alert(user, "[anchored ? "un" : ""]securing...")
+	tool.play_tool_sound(src)
+	if(tool.use_tool(src, user, 6 SECONDS))
+		playsound(loc, 'sound/items/deconstruct.ogg', 50, vary = TRUE)
+		balloon_alert(user, "[anchored ? "un" : ""]secured")
+		deconstruct()
+		return TRUE
+
+/obj/machinery/digital_clock/welder_act(mob/living/user, obj/item/tool)
+	if(user.combat_mode)
+		return
+	if(atom_integrity >= max_integrity)
+		balloon_alert(user, "it doesn't need repairs!")
+		return TRUE
+	balloon_alert(user, "repairing display...")
+	if(!tool.use_tool(src, user, 4 SECONDS, amount = 0, volume=50))
+		return TRUE
+	balloon_alert(user, "repaired")
+	atom_integrity = max_integrity
+	set_machine_stat(machine_stat & ~BROKEN)
+	update_appearance()
+	return TRUE
+
+/obj/machinery/digital_clock/multitool_act(mob/living/user, obj/item/tool)
+	if(user.combat_mode)
+		return
+	if(!(obj_flags & EMAGGED))
+		return
+	balloon_alert(user, "resetting...")
+	tool.play_tool_sound(src)
+	if(tool.use_tool(src, user, 6 SECONDS))
+		playsound(loc, 'sound/items/deconstruct.ogg', 50, vary = TRUE)
+		balloon_alert(user, "reset")
+		obj_flags &= ~EMAGGED
+		return TRUE
+
+/obj/machinery/digital_clock/emag_act(mob/user)
+	if(obj_flags & EMAGGED)
+		return
+	playsound(src, SFX_SPARKS, 100, vary = TRUE, extrarange = SHORT_RANGE_SOUND_EXTRARANGE)
+	do_sparks(3, cardinal_only = FALSE, src)
```

Comment:
Click on Files Changed to see lints for code. You need a named parameter for src because you can't mix kwargs and non-kwargs so easily